pkg/generator/secret_template_reconciler.go

```diff
@@ -202,6 +255,13 @@ func resolveInputResource(ref sg2v1alpha1.InputResourceRef, namespace string, in
 	return toUnstructured(ref.APIVersion, ref.Kind, namespace, resolvedName.String())
 }
 
+// Returns whether we should track the resources contained in a SecretTemplate.
+// We only track resources when a ServiceAccountName has not been specified. This implicitly means
+// we only track Secret resources.
+func isTrackingInputResources(s *sg2v1alpha1.SecretTemplate) bool {
```

Comment:
*suggestion non-blocking*: could this be something like `isTrackingRequired`, I was a bit confused at first

Reply:
I wonder if the `is` is slightly confusing here as `*sg2v1alpha1.SecretTemplate` itself isn't tracking resources. Maybe `shouldTrack` or `shouldTrackInputResources` is better?

Reply:
`shouldTrackInputResources` works for me 👍